src/libstd/thread/mod.rs

```diff
@@ -468,13 +468,35 @@ pub fn current() -> Thread {
 
 /// Cooperatively gives up a timeslice to the OS scheduler.
 ///
+/// This is used when the programmer knows that the thread will have nothing
+/// to do for some time, and thus avoid wasting computing time.
+///
+/// For example when polling on a resource, it is common to check that it is
+/// available, and if not to yield in order to avoid busy waiting. Thus the
+/// poll then yield, is a rather common pattern when implementing low-level
```

Comment:
This sentence is a bit confusing. "the poll then yield" for example is a bit hard to understand with just one reading. Don't hesitate to make the sentence longer if it helps the reader to understand more easily.